bot/log.py

```diff
@@ -14,11 +14,27 @@
 TRACE_LEVEL = 5
 
 
+class CustomLogger(Logger):
```

Comment:
The logger should inherit [.getLoggerClass()](https://docs.python.org/3/library/logging.html?highlight=logging#logging.getLoggerClass) to play nice with other loggers.

Reply:
Changing this to `class CustomLogger(logging.getLoggerClass())` results in PyCharm giving a warning stating `Unresolved attribute reference 'isEnabledFor' for class 'CustomLogger'` and `Unresolved attribute reference '_log' for class 'CustomLogger'` (same happens for `log` instead of `_log`).

Even [`cast`](https://docs.python.org/3/library/typing.html#typing.cast)ing doesn't fix this (`cast(Logger, logging.getLoggerClass())`)

Reply:
> Unresolved attribute reference 'isEnabledFor' for class 'CustomLogger'

This seems to be a PyCharm bug, as both `mypy` and `pyright` correctly infers the correct type.

However, `mypy` (but not `pyright`) returns a different error relating to dynamic base classes. The workaround for that is
```python

if typing.TYPE_CHECKING:
    LoggerClass = Logger
else:
    LoggerClass = logging.getLoggerClass()

class CustomLogger(LoggerClass):
```
A nice explanation of why casting and even type variables will not work for this case can be found here: https://github.com/python/mypy/issues/11050
An even longer read (with a guido appearance) can be found here: https://github.com/python/mypy/issues/2477

Reply:
Committed as part of [ca2d4a5](https://github.com/python-discord/bot/pull/1831/commits/1adae82ebc32fc39636f4ca2d20ec2e2b3c3fbd0)